### PR TITLE
Added support of dynamic bucket resharding for brownfield scenario

### PR DESCRIPTION
**Pull request by udaysk23**
Added support of dynamic bucket resharding for brownfield scenario

Signed-off-by: udaysk23 <ukurundw@redhat.com>

### CODE REVIEW

**Review by viduship — COMMENTED** [review]
Uday, some tox checks are failing, please address that.

Also, could you provide the logs for the same.

**Comment by udaysk23** [comment]
Following is the log location of successful run
http://pastebin.test.redhat.com/1021242

File: rgw/v2/lib/resource_op.py [hunk 1]
```diff
@@ -218,6 +226,7 @@ def read(self):
         self.sts = self.doc["config"].get("sts")
         self.ceph_conf = self.doc["config"].get("ceph_conf")
         self.gc_verification = self.doc["config"].get("gc_verification", False)
+        self.cleanup = self.doc["config"].get("cleanup", False)
```
Comment (psathyan) [inline rgw/v2/lib/resource_op.py:229]: IMO, we should cleanup by default hence the value must be `True`

**Review by psathyan — COMMENTED** [review]
I think this is a good start... some of my thoughts

Changes to `test_Mbuckets_with_Nobjects.py` would suffice, if we can do

```
if config.dbr_scenario == "brownfield":
  dbr_info = utils.read_dbr_info()
  if dbr_info.get("users"):
    all_users_info = deepcopy(dbr_info["users"])
  else:
    all_users_info = s3lib.create_users(config.user_count)
    dbr_info["users"] = deepcopy(all_users_info)
    dbr_info["buckets"] = list()
else:
  all_users_info = s3lib.create_users(config.user_count)

...

if config.dynamic_resharding:
  <perform global configuration changes>
...
if config.dbr_scenario == "brownfield":
  dbr_info["buckets"].append(bucket_name_to_create)
  ... <stats, etc>
```
The `read_dbr_info` method would need to be added to utils
```
def read_dbr_info():
  """
    contents of yaml file
     users: 
       - user_1_response
       - user_2_response
       -
     buckets:
       -
  """
  if not os.path.exists("dbr_info.yaml"):
    return {}

  with open("dbr_info.yaml", "r") as h:
    return yaml.safe_load(h)
```

**Review by psathyan — APPROVED** [review]

**Pull request merged**